### PR TITLE
#675: excluded non-static non-transient pmd rule

### PR DESCRIPTION
**Pull request by vkuchyn**
For issue #675 
Added test to check if class can contain non-static non-transient field. Test failed.
Fixed test - excluded rule `BeanMembersShouldSerialize ` from ruleset.

### CODE REVIEW

**Comment by vkuchyn** [comment]
@davvd please, find someone for review

**Comment by davvd** [comment]
@vkuchyn I will find a reviewer for your pull requests shortly, thanks for contribution!

**Comment by davvd** [comment]
@dmzaytsev please review, thanks

File: qulice-pmd/src/main/resources/com/qulice/pmd/ruleset.xml [hunk 1]
```diff
@@ -32,10 +32,10 @@
  * @version $Id$
  -->
 <ruleset name="Qulice Ruleset"
-    xmlns="http://pmd.sf.net/ruleset/1.0.0"
-    xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
-    xsi:schemaLocation="http://pmd.sf.net/ruleset/1.0.0 http://pmd.sf.net/ruleset_xml_schema.xsd"
-    xsi:noNamespaceSchemaLocation="http://pmd.sf.net/ruleset_xml_schema.xsd">
+         xmlns="http://pmd.sf.net/ruleset/1.0.0"
```
Comment (dmzaytsev) [inline qulice-pmd/src/main/resources/com/qulice/pmd/ruleset.xml]: @vkuchyn why did you change the indentation here and below?

**Comment by dmzaytsev** [comment]
@vkuchyn please see one comment/question above

**Comment by vkuchyn** [comment]
@dmzaytsev sorry, these indentations were made automatically by IDEA. I've fixed them. 
Can you please provide IDEA code style for me to avoid such situations in future.
Also have question - this situation shows that qulice have no reaction on xml code style, maybe I shal report an issue on it?

**Comment by dmzaytsev** [comment]
@vkuchyn thanks
yes, feel free to submit an issue if you have a question or a bug

regarding IDEA settings try to download it [here](http://img.tpc2.com/idea-settings.jar)

**Comment by dmzaytsev** [comment]
@rultor merge

File: qulice-pmd/src/test/java/com/qulice/pmd/PMDValidatorTest.java [hunk 2]
```diff
@@ -491,4 +497,19 @@ public void allowsAssertFail()
         ).validate();
     }
 
+    /**
+     * PMDValidator can allow non-static, non-transient fields.
+     * @throws Exception If something wrong happens inside.
+     */
+    @Test
+    public void allowNonTransientFields() throws Exception {
```
Comment (krzyk) [inline qulice-pmd/src/test/java/com/qulice/pmd/PMDValidatorTest.java]: @vkuchyn ` allows`

File: qulice-pmd/src/test/java/com/qulice/pmd/PMDValidatorTest.java [hunk 3]
```diff
@@ -91,6 +91,12 @@
         "Avoid using Plain JUnit assertions";
 
     /**
+     * Template for using non transient fields.
+     */
+    private static final String NON_TRANSIENT =
```
Comment (krzyk) [inline qulice-pmd/src/test/java/com/qulice/pmd/PMDValidatorTest.java]: @vkuchyn in most cases there is no point in making single use constants (there are cases where they add readability, but not here)

**Comment by krzyk** [comment]
@vkuchyn please see my comments

File: qulice-pmd/src/test/java/com/qulice/pmd/PMDValidatorTest.java (hunk 2 above)
Comment (vkuchyn) [inline qulice-pmd/src/test/java/com/qulice/pmd/PMDValidatorTest.java]: @krzyk I am confused. From one side we have policy of documentation like `Class can do something` and I received recommendations for improvement in [previous ticket](https://github.com/teamed/qulice/pull/665#discussion-diff-51991973). 
Or `PMDValidator can allow`
Or `PMDValidator allows` 
What is correct?

Comment (krzyk) [inline qulice-pmd/src/test/java/com/qulice/pmd/PMDValidatorTest.java]: @vkuchyn The answer is here http://www.yegor256.com/2014/04/27/typical-mistakes-in-java-code.html#test-method-names :)
> It's important to start the first sentence of your JavaDoc with the name of the class you're testing followed by can. So, your first sentence should always be similar to "somebody can do something".

> The method name will state exactly the same, but without the subject. If I add a subject at the beginning of the method name, I should get a complete English sentence, as in above example: "HttpRequest returns its content in Unicode".

Comment (vkuchyn) [inline qulice-pmd/src/test/java/com/qulice/pmd/PMDValidatorTest.java]: @krzyk Now it is clear, thanks.

**Comment by krzyk** [comment]
@rultor merge pls

**Comment by davvd** [comment]
@dmzaytsev Much appreciated! **19 mins** was added to your account, payment ID `77216345`, time spent is 23 hours and 19 mins; extra minutes for review comments (c=4); added +19 to your rating, now it is equal to [+3078](http://www.netbout.com/b/35166?open=rating)

**Comment by davvd** [comment]
@rultor deploy now pls

**Comment by krzyk** [comment]
@davvd this is not urgent

**Comment by davvd** [comment]
> @davvd this is not urgent

@krzyk OK, I un-labeled it as "urgent" tag
